File: Bassett/02_webScraping.py
"""
url : 
    https://kr.indeed.com/jobs?q=python&start=0
    https://kr.indeed.com/jobs?q=python&start=10
    https://kr.indeed.com/jobs?q=python&start=20
    ...
    https://kr.indeed.com/jobs?q=python&start=100

data : 리스트 내역 
result : csv 파일로 저장
"""

from bs4 import BeautifulSoup
import requests


BASE_URL = 'https://kr.indeed.com/jobs?q=python&start='

def makeurl(pagenum):
    return BASE_URL + str(pagenum * 10)


result = []

for i in range(0, 11):
    mUrl = makeurl(i)
    r = requests.get(url=mUrl, timeout=5)
    # print(i, mUrl, r.status_code, '파싱 시작합니다.')
    
    soup = BeautifulSoup(r.text, 'html.parser')
    jobSection = soup.find_all('div', attrs={'class', 'job_seen_beacon'})
    for section in jobSection:
        job = section.find('h2', attrs={'class', 'jobTitle'})
        location = section.find('div', attrs={'class', 'companyLocation'}).text
        dic = dict()
        dic['title'] = job.text
        dic['link '] = f"https://kr.indeed.com/채용보기?jk={job.find('a').attrs['data-jk']}"
        dic['location'] = location
        
        print(i, mUrl, r.status_code, ' ------> 파싱 완료.')
        result.append(dic)
print(result)

File: Bassett/02_webScraping.go
package main

import (
	"fmt"
	"log"
	"net/http"
	"strconv"

	"github.com/PuerkitoBio/goquery"
)

// vscode에서 go.mode file not found ... 오류가 나면
// 아래 터미널 창에서 go env -w GO111MODULE=auto 실행 후 해보시길.

var baseURL = "https://kr.indeed.com/jobs?q=python&start="

type joblist struct {
	jobtitle string
	location string
	link     string
}

func main() {

	for i := 0; i < 10; i++ {
		getPages(i)
	}

}

func getPages(page int) {
	var jobs []joblist
	c := make(chan joblist)
	pageURL := baseURL + strconv.Itoa(page*10)
	res, err := http.Get(pageURL)
	checkErr(err)
	checkCode(res)

	defer res.Body.Close()

	doc, err := goquery.NewDocumentFromReader(res.Body)
	checkErr(err)

	searchData := doc.Find("div.job_seen_beacon")
	searchData.Each(func(i int, s *goquery.Selection) {
		go getJobs(s, c)
		job := <-c
		jobs = append(jobs, job)
	})

	fmt.Println(jobs)
}

func getJobs(card *goquery.Selection, c chan<- joblist) {
	title := card.Find("h2.jobTitle a>span").Text()
	location := card.Find("div.companyLocation").Text()
	datajk, _ := card.Find("a").Attr("data-jk")
	c <- joblist{
		jobtitle: title,
		location: location,
		link:     datajk,
	}
}

func checkErr(err error) {
	if err != nil {
		log.Fatalln(err)
	}
}

func checkCode(res *http.Response) {
	if res.StatusCode != 200 {
		log.Fatalln("error status", res.StatusCode)
	}
}
